### PR TITLE
Adding Spark 2.2 support

### PR DESCRIPTION
**Pull request by akhurana001**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ the `ScheduledSparkApplication` CRD and the corresponding CRD controller.
 
 Spark Operator currently supports the following list of features:
 
-* Supports Spark 2.3 and up.
+* This branch specifically supports Spark 2.2 fork with Kubernetes Support (https://github.com/apache-spark-on-k8s/spark)
```
Comment (kumare3) [inline README.md]: lets add a comment here, that we will be moving to 2.4 when available and 2.4 will deprecate this branch

File: docs/quick-start-guide.md [hunk 2]
```diff
@@ -150,7 +150,13 @@ Events:
 ```
 
 The Spark Operator submits the Spark Pi example to run once it receives an event indicating the `SparkApplication` 
-object was added. 
+object was added.
+
+This operator branch is build on Spark 2.2 fork with Kubernetes which supports PySpark executions. To run the PySpark Spark Pi example, run the following command:
+
+```bash
+$ kubectl apply -f examples/spark-pi-py.yaml
```
Comment (kumare3) [inline docs/quick-start-guide.md]: nice!

File: examples/spark-pi-py.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+spec:
+  type: Python
+  mode: cluster
+  mainApplicationFile: "local:///opt/spark/examples/src/main/python/pi.py"
+  Arguments:
+   - "10"
+  deps:
+    jars:
+      - "local:///opt/spark/examples/jars/spark-examples_2.11-2.2.0-k8s-0.5.0.jar"
+  volumes:
```
Comment (kumare3) [inline examples/spark-pi-py.yaml]: hmm, do e need volume? if so, we should explain if this is a empty dir?

File: examples/spark-pi-py.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        type: Directory
+  driver:
+    cores: 0.1
+    image: kubespark/spark-driver-py:v2.2.0-kubernetes-0.5.0
+    coreLimit: "200m"
+    memory: "512m"
+    labels:
+      version: 2.2.0
+    serviceAccount: spark
+    volumeMounts:
```
Comment (kumare3) [inline examples/spark-pi-py.yaml]: why do we need the same volume on driver and executor? Can we try without volume? Examples are dangerous

Comment (kumare3) [reply examples/spark-pi-py.yaml]: actually i see thats what exists in the original pi example. ignore me. but we should put a comment that volume is not required.

File: pkg/config/constants.go [hunk 5]
```diff
@@ -76,9 +76,9 @@ const (
 	// SparkNodeSelectorKeyPrefix is the configuration property prefix for specifying node selector for the pods.
 	SparkNodeSelectorKeyPrefix = "spark.kubernetes.node.selector."
 	// SparkDriverContainerImageKey is the configuration property for specifying a custom driver container image.
-	SparkDriverContainerImageKey = "spark.kubernetes.driver.container.image"
+	SparkDriverContainerImageKey = "spark.kubernetes.driver.docker.image"
```
Comment (kumare3) [inline pkg/config/constants.go]: why this change?

**Comment by kumare3** [comment]
Couple comments.

File: pkg/config/constants.go (hunk 5 above)
Comment (akhurana001) [reply pkg/config/constants.go]: driver/executor image config keys were modified from 2.2 to 2.3. This is in-sync with 
https://github.com/apache-spark-on-k8s/spark/blob/branch-2.2-kubernetes/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/config.scala#L38

File: README.md (hunk 1 above)
Comment (akhurana001) [reply README.md]: done

File: examples/spark-pi-py.yaml (hunk 4 above)
Comment (akhurana001) [reply examples/spark-pi-py.yaml]: done

File: pkg/config/constants.go (hunk 5 above)
Comment (kumare3) [reply pkg/config/constants.go]: ok, cool

File: examples/spark-pi-py.yaml (hunk 4 above)
Comment (kumare3) [reply examples/spark-pi-py.yaml]: i dont see the comment

**Comment by kumare3** [comment]
LGTM 👍
